### PR TITLE
Infer interface when switching to an non-existing interface

### PR DESCRIPTION
**Pull request by tmattio**
This PR changes the behavior of the switch implementation-interface feature to initialize the interface with an inferred one when switching to a non-existing file.

It depends on https://github.com/ocaml/ocaml-lsp/pull/308.

I implemented a custom request `ocamllsp/inferIntf` in `ocaml-lsp`, because running a codeAction automatically did not seem like a good way to implement this. The code action is still in place though: when switching to an existing interface, we can select some text and replace it with the inferred interface.

### CODE REVIEW

**Comment by rgrinberg** [comment]
This is no longer relevant I believe.

**Comment by tmattio** [comment]
I experimented with `workspace/applyEdit` but did not find a conclusive solution with it, so I believe a custom request remains our best option for this feature. The PR is here: https://github.com/ocaml/ocaml-lsp/pull/314. 

I'll reopen until we decide to merge it, or we have a better solution.

**Comment by tmattio** [comment]
ocaml/ocaml-lsp#314 is merged, so this is good to go if you don't have any concerns.

**Review by mnxn — APPROVED** [review]

**Pull request merged**
